test/periphery/CardPaymentProcessorUpgradeable.test.ts

```diff
@@ -0,0 +1,1421 @@
+      await expect(cardPaymentProcessor.setRevocationCounterMaximum(
+        revocationCounterMaximumDefaultValue
+      )).not.to.emit(
+        cardPaymentProcessor,
+        "SetRevocationCounterMaximum"
+      );
+    });
+  });
+
+  describe("Function 'makePayment()'", async () => {
+    let payment: TestPayment;
+    let authorizationId: string;
+    let correlationId: string;
+
+    beforeEach(async () => {
+      payment = {
+        authorizationId: 123,
+        account: user1,
+        amount: 234,
+        status: PaymentStatus.Nonexistent,
+        makingPaymentCorrelationId: 345,
+      };
+      authorizationId = createBytesString(payment.authorizationId, BYTES16_LENGTH);
+      correlationId = createBytesString(payment.makingPaymentCorrelationId, BYTES16_LENGTH);
+      await setUpContractsForPayments([payment])
+    });
+
+    it("Is reverted if the contract is paused", async () => {
+      await proveTx(cardPaymentProcessor.setPauser(deployer.address));
+      await proveTx(cardPaymentProcessor.pause());
+      await expect(cardPaymentProcessor.connect(payment.account).makePayment(
+        payment.amount,
+        authorizationId,
+        correlationId
+      )).to.be.revertedWith(REVERT_MESSAGE_IF_CONTRACT_IS_PAUSED);
+    });
+
+    it("Is reverted if the payment token amount is zero", async () => {
+      const zeroAmount: number = 0;
+      await expect(cardPaymentProcessor.connect(payment.account).makePayment(
+        zeroAmount,
+        authorizationId,
+        correlationId
+      )).to.be.revertedWith(REVERT_MESSAGE_IF_PAYMENT_AMOUNT_IS_ZERO);
+    });
+
+    it("Is reverted if the payment authorization ID is zero", async () => {
+      await expect(cardPaymentProcessor.connect(payment.account).makePayment(
+        payment.amount,
+        ZERO_BYTES16_STRING,
+        correlationId
+      )).to.be.revertedWith(REVERT_MESSAGE_IF_PAYMENT_AUTHORIZATION_ID_IS_ZERO);
+    });
+
+    it("Is reverted if the user has not enough token balance", async () => {
+      const excessTokenAmount: number = payment.amount + 1;
+      await expect(cardPaymentProcessor.connect(payment.account).makePayment(
+        excessTokenAmount,
+        authorizationId,
+        correlationId
+      )).to.be.revertedWith(REVERT_MESSAGE_IF_TOKEN_TRANSFER_AMOUNT_EXCEEDS_BALANCE);
+    });
+
+    it("Transfers tokens as expected, emits the correct event, changes the state properly", async () => {
+      await checkCardPaymentProcessorState([payment]);
+      await expect(cardPaymentProcessor.connect(payment.account).makePayment(
+        payment.amount,
+        authorizationId,
+        correlationId
+      )).to.changeTokenBalances(
+        brlcMock,
+        [cardPaymentProcessor, payment.account],
+        [+payment.amount, -payment.amount]
+      ).and.to.emit(
+        cardPaymentProcessor,
+        "MakePayment"
+      ).withArgs(
+        authorizationId,
+        correlationId,
+        payment.account.address,
```

Comment:
Codacy has a fix for the issue: [Insert `··`](https://app.codacy.com/gh/cloudwalk/brlc-token/pullRequest?prid=9781596)
```suggestion
          payment.account.address,
```